File: Difficulty: Medium/Find rectangle with corners as 1/find-rectangle-with-corners-as-1.cpp
class Solution {
  public:
    bool ValidCorner(vector<vector<int>>& mat) {
        int r = mat.size();         // number of rows
        int c = mat[0].size();      // number of columns

        unordered_set<string> st;   // to store seen column pairs

        for (int i = 0; i < r; i++) {
            vector<int> cols;       // store column indices where row i has 1

            for (int j = 0; j < c; j++) {
                if (mat[i][j] == 1) {
                    cols.push_back(j);
                }
            }

            // check all pairs of 1s in current row
            for (int x = 0; x < cols.size(); x++) {
                for (int y = x + 1; y < cols.size(); y++) {
                    string valid_pair = to_string(cols[x]) + "_" + to_string(cols[y]);

                    // if this pair already exists in some previous row → rectangle found
                    if (st.count(valid_pair)) {
                        return true;
                    }

                    // insert this pair for the current row
                    st.insert(valid_pair);
                }
            }
        }

        return false;  // no rectangle with 1s found
    }
};

File: Difficulty: Medium/Find rectangle with corners as 1/find-rectangle-with-corners-as-1.java
class Solution {
    public boolean ValidCorner(int[][] mat) {
        int m = mat.length;        // number of rows
        int n = mat[0].length;     // number of columns

        // Try all pairs of rows
        for (int r1 = 0; r1 < m; r1++) {
            for (int r2 = r1 + 1; r2 < m; r2++) {
                // Check for column pairs where both rows have 1
                for (int c1 = 0; c1 < n; c1++) {
                    for (int c2 = c1 + 1; c2 < n; c2++) {
                        // If all four corners are 1s, rectangle exists
                        if (mat[r1][c1] == 1 && mat[r1][c2] == 1 &&
                            mat[r2][c1] == 1 && mat[r2][c2] == 1) {
                            return true;
                        }
                    }
                }
            }
        }

        return false;  // No valid rectangle found
    }
}